baybe/recommenders/pure/nonpredictive/base.py

```diff
@@ -41,24 +41,18 @@ def recommend(
                 f"consider any objectives, meaning that the argument is ignored.",
                 UnusedObjectWarning,
             )
-        if (pending_experiments is not None) and (
-            self.allow_recommending_pending_experiments
-            or searchspace.type is not SearchSpaceType.DISCRETE
-        ):
+        if pending_experiments is not None:
             warnings.warn(
                 f"Pending experiments were provided but the selected recommender "
-                f"'{self.__class__.__name__}' only utilizes this information for "
-                f"purely discrete spaces and "
-                f"{fields(self.__class__).allow_recommending_pending_experiments.name}"
-                f"=False.",
+                f"'{self.__class__.__name__}' does not use this information.",
```

Comment:
@Scienfitz: I remember us having a longer discussion about this part in one of the previous PRs. With the new changes, I think the semantics are now considerable easier but we probably still need to iterate once again to align what the desired behavior should be

Reply:
If I get your points here right, I'd also favor the variant where warnings are always being printed. If people are bothered by this, people can still globally deactivate warnings.
In general, I think that we should als re-visit in general how we handle warnings as I think there are quite some places where people might want to deactivate one type of warning while keeping another one. But this should not be done here :D

Reply:
I dont think the words `pending experimetns` were mentioned once in the meeting, so not sure what you refer to...

Also, the argument that something is only possible if the imagined future with a filter method need to be discarded. We already filter now, just call it toggling, there is no need to defer anything into that imagined future because of it

If we look at the defacto state of the code now, non-pred recommenders can be passed pending experiments and could create this perma warning - This is not good, I remind you of the situation where the simulation module raised thousands of warnings because of the intial data. While its true that it can simply be discarded, users are confused and wonder rightfully whether they did something wrong, but in reality it was just us overly eager raising warnings.

So it seems simply to me the cleanest solution is pending experiments arg needs to be removed for non-pred recommenders. The warning in recommenders becomes obsolete then. The campaign needs to decide whether it passes pending experiments or uses toggling or raises a warning if the respective combo appears - this shifting of responsibilities is akin to what you did with the other flag warnings -  and again I dont see how that should be affected by further changes to renaming/moving our toggling/fitlering in an imaginary future.

Reply:
Your suggested solution is also fine for me 👍🏼 I just don't see how it really makes a difference for you:


> non-pred recommenders can be passed pending experiments and could create this perma warning - This is not good

In your version, you simply turn the warning into an error. So in both cases, it requires the user to "change their code" if the want to get rid of the problem. Why is that better for you?

But let's consider for the moment we agree on doing so: what is your concrete suggestion how to handle the situation on the campaign side?
* Toggling: this actually addresses the "other" of the two concerns. When someone just passed `pending_experiments` they have not all specified whether or not they want to get them as recommendations. That happens via the flag in the campaign level. So isn't that apples and oranges?
* Warning: Ok, but why is a warning legit here and not for the recommender?
* Just "not paassing pending experiments": effectively silently drops one of the things the user has explicitly asked for!

Let me know your thoughts

Reply:
One other idea that just came to my mind as a potential solution: we could also remove `pending_experiments` entirely from the signature, which would solve the trouble with the interface. Instead, we could have a context manager a la 
```python
with recommender.pending_experiments(df):
    recommender.recommend(...)
```
Just some food for thought. Perhaps this opens the path for a clean solution?

Reply:
of course the campaign needs to take are of not passing the pending experiments in cases where they would cause an error, so we do not trade error for warnings

I also thought that maybe we should entirely skip the recommender warning/error by not even having `pending-experiments` for non-predictive recommenders, so I agree that the protocol could change.

If the context manager helps the campaign in doing that job, why not. But the argument should still be present for bayesian recommenders, think eg for the implications i would have for the API etc.

Reply:
> so we do not trade error for warnings

Yes, we would. Because your argument was: 

> non-pred recommenders can be passed pending experiments and could create this perma warning - This is not good

So I can literally just do `RandomRecommender.recommend(batch_size=..., searchspace=..., pending_experiments=...)` as a single call without any context and would get 1) a warning in my version 2) an error in your version.

While changing the protocol is something to consider, I think it's out-of-scope for this PR. Hence, let's think about a clean intermediate solution that does not break later.

Can you give me a concrete suggestion in terms of the three options you mentioned above and then we see if we can get it running in a reasonable way?

Reply:
I dont want to suggest a verison where an error is triggered
this is achieved by the campaign taking care of providing `pending_experiments` or not, something that doesnt seem to appear in the suggestions above

I think that should be independent of the protocol change. Is the protocol actually a big deal? is it ok if a class implementing the protocol "adds" an argument to the method? then is should be rather trivial

Reply:
does it make sense to you?

- let non-predictive recommenders throw an error if provided pending expeirments
- make campaign take care that no such error is actually thrown if `recommend` is called via campaign. the campaign can decide its warning/error logic based on the old flags just as the logic was inside recommenders before
- the protocol change is not necessary for that

Reply:
Do you need an additional opinion here? This discussion goes on for quite some time already. Or do we want to discuss this in our Meeting this week?

Reply:
Let's align once and for all in the dev meeting 🙃